### PR TITLE
[FEAT] [KV] expose stream `metadata`

### PR DESCRIPTION
**Pull request by aricart**

### CODE REVIEW

File: jetstream/kv.ts [hunk 1]
```diff
@@ -917,6 +920,10 @@ export class KvStatusImpl implements KvStatus {
   get size(): number {
     return this.si.state.bytes;
   }
+
+  get metadata(): Record<string, string> | undefined {
+    return this.si.config.metadata;
```
Comment (caspervonb) [inline jetstream/kv.ts:925]: Sure you don't want to default to an empty object when undefined? less cognitive overload on the caller.

Comment (caspervonb) [reply jetstream/kv.ts:925]: Also looks like object store does return ?? empty.

File: jetstream/kv.ts [hunk 2]
```diff
@@ -917,6 +920,10 @@ export class KvStatusImpl implements KvStatus {
   get size(): number {
     return this.si.state.bytes;
   }
+
+  get metadata(): Record<string, string> | undefined {
```
Comment (caspervonb) [inline jetstream/kv.ts:924]:
```suggestion
  get metadata(): Record<string, string> {
```

File: jetstream/kv.ts (hunk 1 above)
Comment (aricart) [reply jetstream/kv.ts:925]: you are absolutely correct

**Review by caspervonb — APPROVED** [review]
lgtm

**Pull request merged**